models/classes/runner/navigation/QtiRunnerNavigation.php

```diff
@@ -101,41 +108,72 @@ public static function move($direction, $scope, RunnerServiceContext $context, $
      * @param RunnerServiceContext $context
      * @param int $nextPosition
      */
-    public static function checkTimedSectionExit(RunnerServiceContext $context, $nextPosition)
+    public static function checkTimedSectionExit(RunnerServiceContext $context, $nextPosition): void
     {
         $timerConfig = $context->getTestConfig()->getConfigValue('timer');
+
         if (empty($timerConfig['keepUpToTimeout'])) {
             /* @var AssessmentTestSession $session */
             $session = $context->getTestSession();
             $route = $session->getRoute();
             $section = $session->getCurrentAssessmentSection();
-            $limits = $section->getTimeLimits();
-
-            // As we have only one identifier for the whole adaptive section it will consider a jump of section on the
-            // first item
-            if (!($context instanceof QtiRunnerServiceContext) || !$context->isAdaptive()) {
```

Comment:
I am a bit concerned in generate the size of this refactoring and all critical parts touched, knowing we have such small test coverage for this. 

Please, be sure all we touched will continue to work as it was before or revert the refactoring if not confident.

Reply:
Some changes were actually done because the pipeline checks were complaining about this method size, and refactoring it seemed like a quick win. Other changes were done just because it helped with readability while debugging.

I can revert them if you prefer to avoid the risk.

Reply:
I trust you, if you think it is low risk. I did not find test case for this method in the unit tests. That was my main concern.